RELEASE.md

```diff
@@ -0,0 +1,18 @@
+# Release Process
+
+1. Send announcement that a release is underway to the committer's lounge on
+discourse.numenta.org and ask reviewers not to merge PRs in NuPIC until you're
+done with the release.
+2. Create a PR that includes:
+    - Release notes added to CHANGELOG.md
+    - Change to the VERSION file so it matches the intended release version
+3. Wait for the PR to be approved and merged, and the Bamboo build to complete
+successfully
+4. Create a "release" in Bamboo with a version matching the intended release
```

Comment:
BTW, I think we can do away with this step at some point if we can figure out how to make the contents of `VERSION.txt` available as a [Bamboo deployment variable](https://confluence.atlassian.com/bamboo/variables-for-deployment-environments-342754180.html).